editor/src/clj/editor/ui.clj

```diff
@@ -750,14 +753,43 @@
   (value [this] (.getValue this))
   (value! [this val] (with-on-edit-event-suppressed! this (.setValue this val))))
 
+(declare bind-key!)
+
 (extend-type TextField
   HasAction
-  (on-action! [node fn]
+  (on-action! [node update-fn]
     (.setOnAction node (event-handler e
                          ;; Clear the auto-commit flag. Further edits will re-apply it.
                          (when (user-data node ::auto-commit)
                            (user-data! node ::auto-commit false))
-                         (fn e)))))
+                         (if (zero? (.getLength (.getSelection node)))
+                           (.selectAll node)
+                           (.deselect node))
+                         (update-fn e))))
+  Cancellable
+  (on-cancel! [node cancel-fn]
+    (bind-key! node "Esc" (fn []
+                            (cancel-fn node)
+                            (when-let [parent (.getParent node)]
+                              (.requestFocus parent))))))
+
+(extend-type TextArea
```

Comment:
extend TextArea using the same protocols we use for text field, but taken into account implementation details